dashboard/src/actions/tocActions.js

```diff
@@ -0,0 +1,124 @@
+import * as TYPES from "./types";
+
+import API from "utils/axiosInstance";
+import { DANGER } from "assets/constants/toastConstants";
+import { showToast } from "./toastActions";
+import { uriTemplate } from "utils/helper";
+
+/**
+ * Function to fetch contents data
+ * @function
+ * @param {String} param - Dataset ID
+ * @param {String} dataUri - URI
+ * @param {String} item - Active item
+ * @param {Boolean} isSubDir - To identify sub-directory expansion
+ * @return {Function} - dispatch the action and update the state
+ */
+export const fetchTOC =
+  (param, dataUri, item, isSubDir) => async (dispatch, getState) => {
+    try {
+      dispatch({ type: TYPES.LOADING });
+      const endpoints = getState().apiEndpoint.endpoints;
+      const parent = dataUri?.split("contents/").pop();
+      const uri = uriTemplate(endpoints, "datasets_contents", {
+        dataset: param,
+        target: parent,
+      });
+      const response = await API.get(uri);
+      if (response.status === 200 && response.data) {
+        if (!isSubDir) {
+          dispatch({
+            type: TYPES.SET_INVENTORY_LINK,
+            payload: response.data.uri.replace("contents", "inventory"),
+          });
+        }
```

Comment:
I guess the intent here is to save the link to download the tarball? That's the only place where it really makes sense to convert a "contents" URI to an "inventory" URI. Could this be done in the caller before building the tree? And it could be done more cleanly, then, as `uriTemplate(endpoints, "datasets_inventory"), {dataset: resource_id, target: ""});` right before calling the action method.

I think that then you might not need the `isSubDir` parameter, because in `parseToTreeView` I *think* that's the same as `if (activeItem != "")` or just `if (activeItem)`?